MIME4j-268 MIME4J-269 Convenience helpers...

chibenwa opened the pull request:
- Expose a LENIENT default config
 - Provide a helper method for Message to byte[] conversion

chibenwa committed:
MIME4J-269 Introduce a Lenient default mimeConfig object

chibenwa committed:
MIME4J-268 DefaultMessageWriter should expose a convenient *asBytes* method

aduprat commented on core/src/main/java/org/apache/james/mime4j/stream/MimeConfig.java:
```diff
@@ -26,6 +26,12 @@
  */
 public final class MimeConfig {
 
+    private static final MimeConfig LENIENT = MimeConfig.custom()
```
Why private?

chibenwa replied:
Oups yes should of course be public....

mbaechler replied:
It's not only lenient (and btw you don't even call `setStringParsing(false)` ) but dangerous : max values are here to protect use from malicious input.
What happens with a mail crafted to overflow the parser ? Can I trigger JVM OutOfMemoryException with a single mail ?

chibenwa replied:
I guess so... 

Then maybe I should drop MIME4J-269, and introduce a ticket for defining in details constants to be used safely, while not failing on legitimate email, that can be used by James.

What do you think?

mbaechler replied:
Not sure to understand

chibenwa replied:
I mean, as you mention, dropping all limits to have no input limits is unsafe.

We'd better use a mime-config carefully defining a "safe enough/lenient enough" limit instead that we should be using in James.

mbaechler replied:
agree

chibenwa replied:
I propose 100 M as a safe upper bound for **maximum content length** and keep the other ones unlimited.

This should be enough to not overwhelm the JVM too much while still being very permissive.

We can call such a constant "PERMISSIVE".

WDYT?

mbaechler replied:
why not, needs some testing to check what the actual memory consumption is.

chibenwa committed:
fixup! MIME4J-269 Introduce a Lenient default mimeConfig object

chibenwa committed:
MIME4J-269 Expose a convenient 100 Mo limited "permissive" mime config limit

This aims at remaining very permissive while still denying a single email to use all JVM memory.

rouazana commented on dom/src/test/java/org/apache/james/mime4j/dom/LargeMessageParsingTest.java:
```diff
@@ -0,0 +1,48 @@
+        DefaultMessageBuilder messageBuilder = new DefaultMessageBuilder();
+        messageBuilder.setMimeEntityConfig(MimeConfig.PERMISSIVE);
+        messageBuilder.parseMessage(new ByteArrayInputStream(outputStream.toByteArray()));
+    }
```
can you add some tests with (very) long lines please?

chibenwa committed:
fixup! MIME4J-269 Expose a convenient 100 Mo limited "permissive" mime config limit

chibenwa commented:
Merged
